.github/workflows/android.yml

```diff
@@ -360,19 +364,113 @@ jobs:
             echo "Pulling unit test results..."
             adb pull /data/local/tmp/ov_test/unit_test_results.xml ./unit_test_results.xml || { echo "Warning: Could not pull unit test results"; adb shell "ls -la /data/local/tmp/ov_test/"; adb shell "cat /data/local/tmp/ov_test/unit_test_results.xml" || echo "No unit test results file"; }
 
-      - name: Upload test results
+      - name: Upload unit test results
         uses: actions/upload-artifact@330a01c490aca151604b8cf639adc76d48f6c5d4 # v5.0.0
         if: always()
         with:
-          name: cpu_tests_results_android_x64
+          name: cpu_unit_tests_results_android_x64
+          path: ./unit_test_results.xml
+          if-no-files-found: 'ignore'
+
+  CXX_Func_Tests:
+    needs: [Smart_CI, Docker, Build]
+    timeout-minutes: 120
```

Comment:
Isn't it too generous? The precommit **test** jobs should not run for more than an hour at most.